### PR TITLE
Eclipse 4.8.0 (JDT)

### PR DESCRIPTION
**Pull request by fvgh**
Made Spotless-Eclipse-JDT 4.8.0 default version.
Added functionality to lookup Maven Coordinate information from Jar-State as @nedtwigg proposed in #234.
This PR fixes issues #226 and #191.

### CODE REVIEW

**Comment by fvgh** [comment]
@nedtwigg Not so sure about the change record. Should all CHANGE.md files refer this PR? Should Spotless CHANGE.md also mention #226?

**Comment by nedtwigg** [comment]
> Should Spotless CHANGE.md also mention #226?

Your call.

File: lib-extra/src/main/java/com/diffplug/spotless/extra/java/EclipseJdtFormatterStep.java [hunk 1]
```diff
@@ -43,9 +45,15 @@ public static EclipseBasedStepBuilder createBuilder(Provisioner provisioner) {
...
+		Class<?> formatterClazz = getClass(state);
 		Object formatter = formatterClazz.getConstructor(Properties.class).newInstance(state.getPreferences());
 		Method method = formatterClazz.getMethod(FORMATTER_METHOD, String.class);
 		return input -> (String) method.invoke(formatter, input);
 	}
+
+	private static Class<?> getClass(State state) {
+		if (state.getMavenCoordinate(MAVEN_GROUP_ARTIFACT).isPresent())
+			return state.loadClass(FORMATTER_CLASS);
+		return state.loadClass(FORMATTER_CLASS_OLD);
```
Comment (nedtwigg) [inline lib-extra/src/main/java/com/diffplug/spotless/extra/java/EclipseJdtFormatterStep.java]: Ifs without brackets are dangerous.  I would change to this:

```java
if ( ) {
   return
} else {
   return
}
```

File: lib/src/main/java/com/diffplug/spotless/JarState.java [hunk 2]
```diff
@@ -42,8 +42,7 @@
 public final class JarState implements Serializable {
 	private static final long serialVersionUID = 1L;
 
-	@SuppressWarnings("unused")
-	private final Set<String> mavenCoordinates;
+	private final TreeSet<String> mavenCoordinates;
```
Comment (nedtwigg) [inline lib/src/main/java/com/diffplug/spotless/JarState.java]: Sorry, I almost missed this one.  There are corner-cases where classpath order matters.  Unless we have a really compelling reason, I think we should keep this as `Set` and not `TreeSet`

Comment (fvgh) [reply lib/src/main/java/com/diffplug/spotless/JarState.java]: Just wanted to emphasize that I expect a deterministic result for getMavenCoordinates for the same input. I do not insist. What's your final decision?

Comment (nedtwigg) [reply lib/src/main/java/com/diffplug/spotless/JarState.java]: Set

Comment (nedtwigg) [reply lib/src/main/java/com/diffplug/spotless/JarState.java]: then feel free to press the merge button, and I'll test the `master` integration build on some of my projects

Comment (fvgh) [reply lib/src/main/java/com/diffplug/spotless/JarState.java]: Ok, THX.

**Pull request merged**